packages/core/test/unit/application.test.ts

```diff
@@ -8,10 +8,60 @@ import {Application, Server, Component} from '../../index';
 import {Context, Constructor} from '@loopback/context';
 
 describe('Application', () => {
+  describe('controller binding', () => {
+    let app: Application;
+    class MyController {}
+
+    beforeEach(givenApp);
+
+    it('binds a controller', () => {
+      const binding = app.controller(MyController);
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.MyController');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
```

Comment:
If this assertion fails, the error message will be most likely not helpful. E.g. if MyController ends up at the index 1 of the array returned by `findByTag`, then the error message won't tell us.

I am proposing the following improvement:

```ts
expect(app.findByTag('controller').map(b => b.key))
  .to.containEql('controllers.MyController');
```